module/zfs/spa.c

```diff
@@ -3327,6 +3339,87 @@ spa_start_livelist_condensing_thread(spa_t *spa)
 	    spa_livelist_condense_cb, spa, minclsyspri);
 }
 
+static int
+spa_rrd_write(spa_t *spa, dmu_tx_t *tx, rrd_t *rrd, const char *what,
+    boolean_t insert)
+{
+
+	if (insert) {
+		return (zap_add(spa_meta_objset(spa), DMU_POOL_DIRECTORY_OBJECT,
+		    what, RRD_ENTRY_SIZE, RRD_STRUCT_ELEM, rrd, tx));
+	}
+
+	return (zap_update(spa_meta_objset(spa),
+	    DMU_POOL_DIRECTORY_OBJECT, what, RRD_ENTRY_SIZE, RRD_STRUCT_ELEM,
+	    rrd, tx));
+}
+
+static void
+spa_sync_time_logger(spa_t *spa, uint64_t txg)
+{
+	uint64_t curtime;
+	boolean_t insert;
+	dmu_tx_t *tx;
+
+	if (!spa_writeable(spa)) {
+		return;
+	}
+	curtime = gethrestime_sec();
+	if (curtime < spa->spa_last_noted_txg_time + spa_note_txg_time) {
+		return;
+	}
+
+	spa->spa_last_noted_txg_time = curtime;
+	spa->spa_last_noted_txg = txg;
+
+	mutex_enter(&spa->spa_txg_log_time_lock);
+	dbrrd_add(&spa->spa_txg_log_time, curtime, txg);
+
+	if (curtime < spa->spa_last_flush_txg_time + spa_flush_txg_time) {
+		goto out;
```

Comment:
I wonder if it would make sense to flush on pool export, similar to how we flush log space map.